src/modules/slugs/early_hooks.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+add_filter( 'qtranslate_language_detect_redirect', 'qtranxf_slugs_language_detect_redirect', 600, 3 );
+
+/**
+* Allows url redirection due to language auto detection only if site url has been requested
+*
+* @see qtranxf_check_url_maybe_redirect
+* @param string $url_lang proposed target URL for the active language to redirect to.
+* @param string $url_orig original URL supplied to browser, which needs to be standardized.
+* @param array $url_info a hash of various information parsed from original URL, cookies and other site configuration. The key names should be self-explanatory.
+*
+* @return string resulting redirection url
+*/
+function qtranxf_slugs_language_detect_redirect($url_lang, $url_orig, $url_info): string {
+   global $q_config;
+   if (site_url().'/' === $url_orig)
+       return $url_lang;
+   else {
+       if ( empty( $url_info['lang_url'] ) ){
+           return qtranxf_convertURL( $url_orig, $q_config['default_language'],false,true );
+       } else {
+           return $url_orig;
+       }
+   }
```

Comment:
This doesn't seem to involve any slugs underlying functionality? Why is that needed for slugs and not the core part?

Reply:
This is needed with slugs module to avoid language redirections to a language that can be not consistent with the one the requested slug is referring to. In that case it would result in 404.
The core part is not affected by this as the slug is constant for all languages.
Exception is made for site_url, where no slug is requested.

Reply:
OK then let's keep this but I think we need a special init case for slugs.